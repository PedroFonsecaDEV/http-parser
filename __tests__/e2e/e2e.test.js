const run = require('../../service/run');

describe('e2e test urlpilgrim', () => {
  test('prints help message with arguments: -h', async () => {
    const { stderr, stdout, exitCode } = await run('-h');
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });

  /*  
  > The following tests require managing chalk colors on snapshots. 
  > https://github.com/chalk/supports-color

  test('prints error when no arguments given', async () => {
    const { stderr, stdout, exitCode } = await run();
    expect(exitCode).toBe(1);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });

  test('prints good and bad against the file: url_e2e_test.txt', async () => {
    const { stderr, stdout, exitCode } = await run('-f', 'url_e2e_test.txt');
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  }); */
});
