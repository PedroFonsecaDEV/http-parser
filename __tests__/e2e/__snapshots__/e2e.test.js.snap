// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e test urlpilgrim prints help message with arguments: -h 1`] = `
"Usage: http-parser [option]

Options:
  -f, --file     Please provide a filename                              [string]
  -i, --ignore   Please provide a file with URLs to ignore.             [string]
  -t, --time     Provide ms for waiting for a request                   [number]
  -l, --local    Provide a response from a local server
  -a, --all
  -g, --good
  -b, --bad
  -h, --help     Show usage information & exit                         [boolean]
  -v, --version  Show version number                                   [boolean]"
`;
